File: src/demo-sources/matlab/rayleigh_scattering.m
function rayleigh_scattering()
% 瑞利散射演示 - 为什么天空是蓝色？(Rayleigh Scattering - Why is the Sky Blue?)
% ============================================================================
%
% 演示瑞利散射如何产生蓝天和日落的颜色，以及散射光的偏振特性。
%
% 物理原理:
% ---------
% 瑞利散射强度公式:
%     I(θ, λ) ∝ (1 + cos²θ) / λ⁴
%
%     θ: 散射角度
%     λ: 光波波长
%
% 关键结论:
%     1. 散射强度 ∝ 1/λ⁴（短波长散射更强）
%     2. 蓝光（450nm）散射强度 ≈ 红光（650nm）的 5.6倍
%     3. 90°散射光为完全线偏振
%     4. 日落时光程长，蓝光散射殆尽，红光穿透
%
% 应用:
%     - 大气光学现象解释
%     - 偏振天光导航
%     - 天文观测时间选择
%     - 摄影偏振滤镜使用
%
% 兼容性: MATLAB R2016b+, GNU Octave 4.0+
% 依赖: 无需额外工具箱
% 运行: octave --no-gui rayleigh_scattering.m 或在MATLAB中直接运行
%
% 作者: PolarCraft Team
% 日期: 2026-01-14

    fprintf('==========================================================\n');
    fprintf('瑞利散射演示 - 为什么天空是蓝色？\n');
    fprintf('Rayleigh Scattering - Why is the Sky Blue?\n');
    fprintf('==========================================================\n');
    fprintf('\n物理原理:\n');
    fprintf('  I(θ, λ) ∝ (1 + cos²θ) / λ⁴\n');
    fprintf('  蓝光散射强度 ≈ 红光的 5.6倍\n\n');

    % 参数
    params = struct();
    params.sun_angle = 30;   % 太阳高度角（0=地平线，90=天顶）
    params.view_angle = 90;  % 观察角度

    % 创建图形窗口
    fig = figure('Name', '瑞利散射演示 - Rayleigh Scattering', ...
                 'Position', [100, 100, 1200, 800], ...
                 'Color', [0.06, 0.09, 0.16], ...
                 'NumberTitle', 'off');

    % 创建控件
    controls = create_controls(fig, params);

    % 初始绘图
    update_plot(fig, params, controls);
end

function controls = create_controls(fig, params)
    % 创建交互控件

    % 太阳高度角滑块
    uicontrol('Parent', fig, 'Style', 'text', ...
              'String', sprintf('Sun Elevation (°): %.0f', params.sun_angle), ...
              'Position', [50, 100, 350, 25], ...
              'BackgroundColor', [0.12, 0.16, 0.23], ...
              'ForegroundColor', [0.98, 0.75, 0.14], ...
              'FontSize', 10, 'FontWeight', 'bold', ...
              'HorizontalAlignment', 'left', ...
              'Tag', 'sun_label');

    uicontrol('Parent', fig, 'Style', 'slider', ...
              'Min', 0, 'Max', 90, 'Value', params.sun_angle, ...
              'Position', [50, 80, 350, 20], ...
              'BackgroundColor', [0.12, 0.16, 0.23], ...
              'Callback', {@on_slider_changed, fig, params, 'sun'}, ...
              'Tag', 'sun_slider');

    % 观察角度滑块
    uicontrol('Parent', fig, 'Style', 'text', ...
              'String', sprintf('View Angle (°): %.0f', params.view_angle), ...
              'Position', [50, 50, 350, 25], ...
              'BackgroundColor', [0.12, 0.16, 0.23], ...
              'ForegroundColor', [0.13, 0.83, 0.93], ...
              'FontSize', 10, 'FontWeight', 'bold', ...
              'HorizontalAlignment', 'left', ...
              'Tag', 'view_label');

    uicontrol('Parent', fig, 'Style', 'slider', ...
              'Min', 0, 'Max', 180, 'Value', params.view_angle, ...
              'Position', [50, 30, 350, 20], ...
              'BackgroundColor', [0.12, 0.16, 0.23], ...
              'Callback', {@on_slider_changed, fig, params, 'view'}, ...
              'Tag', 'view_slider');

    controls = struct();
end

function on_slider_changed(src, ~, fig, params, slider_type)
    % 滑块回调
    value = get(src, 'Value');

    if strcmp(slider_type, 'sun')
        params.sun_angle = value;
        label = findobj(fig, 'Tag', 'sun_label');
        set(label, 'String', sprintf('Sun Elevation (°): %.0f', value));
    elseif strcmp(slider_type, 'view')
        params.view_angle = value;
        label = findobj(fig, 'Tag', 'view_label');
        set(label, 'String', sprintf('View Angle (°): %.0f', value));
    end

    update_plot(fig, params, []);
end

function intensity = rayleigh_intensity(wavelength, theta_deg)
    % 计算瑞利散射强度
    % I ∝ (1 + cos²θ) / λ⁴
    theta_rad = deg2rad(theta_deg);
    angular_factor = (1 + cos(theta_rad).^2);
    wavelength_factor = (450 ./ wavelength) .^ 4;  % 归一化到450nm
    intensity = angular_factor .* wavelength_factor;
end

function rgb = wavelength_to_rgb(wavelength)
    % 波长转RGB颜色
    wavelength = max(380, min(750, wavelength));

    if wavelength >= 380 && wavelength < 440
        R = -(wavelength - 440) / (440 - 380);
        G = 0.0;
        B = 1.0;
    elseif wavelength >= 440 && wavelength < 490
        R = 0.0;
        G = (wavelength - 440) / (490 - 440);
        B = 1.0;
    elseif wavelength >= 490 && wavelength < 510
        R = 0.0;
        G = 1.0;
        B = -(wavelength - 510) / (510 - 490);
    elseif wavelength >= 510 && wavelength < 580
        R = (wavelength - 510) / (580 - 510);
        G = 1.0;
        B = 0.0;
    elseif wavelength >= 580 && wavelength < 645
        R = 1.0;
        G = -(wavelength - 645) / (645 - 580);
        B = 0.0;
    else  % 645 <= wavelength <= 750
        R = 1.0;
        G = 0.0;
        B = 0.0;
    end

    % 亮度因子
    if wavelength >= 380 && wavelength < 420
        factor = 0.3 + 0.7 * (wavelength - 380) / (420 - 380);
    elseif wavelength >= 420 && wavelength < 700
        factor = 1.0;
    else  % 700 <= wavelength <= 750
        factor = 0.3 + 0.7 * (750 - wavelength) / (750 - 700);
    end

    rgb = [R * factor, G * factor, B * factor];
end

function [sky_color, sun_color] = calculate_sky_color(sun_elevation)
    % 根据太阳高度计算天空颜色
    path_length_factor = 1.0 / (sin(deg2rad(sun_elevation + 5)) + 0.1);

    wavelengths = [450, 550, 650];  % B, G, R
    intensities = [1.0, 0.6, 0.3];  % 初始强度

    % 瑞利散射衰减
    scattered = intensities .* rayleigh_intensity(wavelengths, 90);
    transmitted = intensities .* exp(-path_length_factor * scattered / 10);

    % 归一化
    sky_color = scattered / max(scattered);
    sun_color = transmitted / max(transmitted);
end

function update_plot(fig, params, ~)
    % 更新所有图形

    % 清除旧图形
    delete(findall(fig, 'Type', 'axes'));

    % 创建天空场景
    ax_sky = axes('Parent', fig, 'Position', [0.05, 0.35, 0.6, 0.55], ...
                  'Color', [0.12, 0.16, 0.23], 'XColor', 'none', 'YColor', 'none');
    hold(ax_sky, 'on');
    xlim(ax_sky, [0, 10]);
    ylim(ax_sky, [0, 8]);
    axis(ax_sky, 'equal');

    draw_sky_scene(ax_sky, params);

    % 创建光谱图
    ax_spectrum = axes('Parent', fig, 'Position', [0.7, 0.55, 0.25, 0.3], ...
                       'Color', [0.12, 0.16, 0.23], ...
                       'XColor', [1, 1, 1], 'YColor', [1, 1, 1]);
    hold(ax_spectrum, 'on');

    draw_spectrum(ax_spectrum, params);

    % 创建极坐标散射图
    ax_polar = axes('Parent', fig, 'Position', [0.7, 0.15, 0.25, 0.35]);
    hold(ax_polar, 'on');

    draw_polar_pattern(ax_polar);
end

function draw_sky_scene(ax, params)
    % 绘制天空场景

    % 天空颜色
    [sky_color, sun_color] = calculate_sky_color(params.sun_angle);

    % 背景天空
    rectangle('Position', [0, 3, 10, 5], ...
              'FaceColor', [sky_color, 0.3], ...
              'EdgeColor', 'none', 'Parent', ax);

    % 地面
    rectangle('Position', [0, 0, 10, 3], ...
              'FaceColor', [0.18, 0.31, 0.09], ...
              'EdgeColor', 'none', 'Parent', ax);

    % 太阳
    sun_rad = deg2rad(params.sun_angle);
    sun_x = 2 + 6 * cos(sun_rad + pi/2);
    sun_y = 3 + 5 * sin(sun_rad + pi/2);

    rectangle('Position', [sun_x - 0.4, sun_y - 0.4, 0.8, 0.8], ...
              'Curvature', [1, 1], 'FaceColor', sun_color, ...
              'EdgeColor', 'none', 'Parent', ax);

    % 太阳光线
    for i = 1:8
        angle = (i - 1) * 45;
        ang_rad = deg2rad(angle);
        dx = 0.7 * cos(ang_rad);
        dy = 0.7 * sin(ang_rad);
        plot([sun_x, sun_x + dx], [sun_y, sun_y + dy], '-', ...
             'Color', sun_color, 'LineWidth', 2, 'Parent', ax);
    end

    % 观察者
    obs_x = 5;
    obs_y = 3;
    rectangle('Position', [obs_x - 0.2, obs_y - 0.2, 0.4, 0.4], ...
              'Curvature', [1, 1], 'FaceColor', [1, 1, 1], ...
              'EdgeColor', 'none', 'Parent', ax);
    text(obs_x, obs_y - 0.5, 'Observer', 'HorizontalAlignment', 'center', ...
         'Color', [1, 1, 1], 'FontSize', 9, 'Parent', ax);

    % 散射光路
    scatter_x = obs_x + 2 * cos(deg2rad(params.view_angle));
    scatter_y = obs_y + 2 * sin(deg2rad(params.view_angle));

    % 箭头（使用线条模拟）
    plot([scatter_x, obs_x], [scatter_y, obs_y], '->', ...
         'Color', [0.13, 0.83, 0.93], 'LineWidth', 2, 'Parent', ax);

    % 散射角度标注
    scattering_angle = abs(params.view_angle - params.sun_angle);
    text(5, 7, sprintf('Scattering Angle: %.0f°', scattering_angle), ...
         'HorizontalAlignment', 'center', 'Color', [0.13, 0.83, 0.93], ...
         'FontSize', 11, 'FontWeight', 'bold', 'Parent', ax);

    % 偏振信息
    polarization = 100 * sin(deg2rad(scattering_angle))^2;
    text(5, 6.5, sprintf('Polarization: %.0f%%', polarization), ...
         'HorizontalAlignment', 'center', 'Color', [0.96, 0.45, 0.71], ...
         'FontSize', 10, 'Parent', ax);
end

function draw_spectrum(ax, params)
    % 绘制光谱散射强度

    wavelengths = linspace(380, 750, 100);
    intensities = rayleigh_intensity(wavelengths, params.view_angle);

    % 绘制光谱条
    for i = 1:(length(wavelengths) - 1)
        wl = wavelengths(i);
        color = wavelength_to_rgb(wl);
        intensity = intensities(i);
        bar(ax, wl, intensity, 4, 'FaceColor', color, ...
            'EdgeColor', 'none', 'FaceAlpha', 0.8);
    end

    % 标注关键波长
    blue_wl = 450;
    red_wl = 650;
    blue_int = rayleigh_intensity(blue_wl, params.view_angle);
    red_int = rayleigh_intensity(red_wl, params.view_angle);
    ratio = blue_int / red_int;

    plot(ax, blue_wl, blue_int, 'o', 'MarkerSize', 10, ...
         'MarkerFaceColor', 'blue', 'MarkerEdgeColor', 'blue');
    plot(ax, red_wl, red_int, 'o', 'MarkerSize', 10, ...
         'MarkerFaceColor', 'red', 'MarkerEdgeColor', 'red');

    xlabel(ax, 'Wavelength (nm)', 'Color', [1, 1, 1]);
    ylabel(ax, 'Scattering Intensity', 'Color', [1, 1, 1]);
    title(ax, sprintf('Blue/Red Ratio: %.2f×', ratio), ...
          'Color', [1, 1, 1], 'FontSize', 10);
    grid(ax, 'on');
    set(ax, 'GridColor', [1, 1, 1], 'GridAlpha', 0.2);
end

function draw_polar_pattern(ax)
    % 绘制极坐标散射图案

    theta = linspace(0, 2*pi, 100);
    r = 1 + cos(theta).^2;  % Rayleigh散射角度分布

    % 转换为极坐标
    polarplot(ax, theta, r, 'Color', [0.13, 0.83, 0.93], 'LineWidth', 2.5);
    hold(ax, 'on');

    % 标注90°散射
    polarplot(ax, [pi/2, pi/2], [0, 1 + cos(pi/2)^2], 'r--', 'LineWidth', 2);
    polarplot(ax, [3*pi/2, 3*pi/2], [0, 1 + cos(3*pi/2)^2], 'r--', 'LineWidth', 2);

    title(ax, 'Scattering Pattern (1 + cos²θ)', ...
          'Color', [1, 1, 1], 'FontSize', 10);
    set(ax, 'Color', [0.12, 0.16, 0.23]);
    ax.ThetaColor = [1, 1, 1];
    ax.RColor = [1, 1, 1];
    grid(ax, 'on');
    set(ax, 'GridColor', [1, 1, 1], 'GridAlpha', 0.3);
end

File: src/demo-sources/python/rayleigh_scattering.py
#!/usr/bin/env python3
"""
瑞利散射演示 - 为什么天空是蓝色？(Rayleigh Scattering - Why is the Sky Blue?)
============================================================================

演示瑞利散射如何产生蓝天和日落的颜色，以及散射光的偏振特性。

物理原理:
---------
瑞利散射强度公式:
    I(θ, λ) ∝ (1 + cos²θ) / λ⁴
    
    θ: 散射角度
    λ: 光波波长
    
关键结论:
    1. 散射强度 ∝ 1/λ⁴（短波长散射更强）
    2. 蓝光（450nm）散射强度 ≈ 红光（650nm）的 5.6倍
    3. 90°散射光为完全线偏振
    4. 日落时光程长，蓝光散射殆尽，红光穿透

应用:
    - 大气光学现象解释
    - 偏振天光导航
    - 天文观测时间选择
    - 摄影偏振滤镜使用

依赖: pip install numpy matplotlib
运行: python rayleigh_scattering.py

作者: PolarCraft Team
日期: 2026-01-14
"""

import numpy as np
import matplotlib.pyplot as plt
from matplotlib.widgets import Slider
from matplotlib.patches import Circle, Wedge, FancyArrowPatch

def wavelength_to_rgb(wavelength):
    """波长转RGB颜色"""
    if wavelength < 380:
        wavelength = 380
    elif wavelength > 750:
        wavelength = 750
    
    if 380 <= wavelength < 440:
        R = -(wavelength - 440) / (440 - 380)
        G = 0.0
        B = 1.0
    elif 440 <= wavelength < 490:
        R = 0.0
        G = (wavelength - 440) / (490 - 440)
        B = 1.0
    elif 490 <= wavelength < 510:
        R = 0.0
        G = 1.0
        B = -(wavelength - 510) / (510 - 490)
    elif 510 <= wavelength < 580:
        R = (wavelength - 510) / (580 - 510)
        G = 1.0
        B = 0.0
    elif 580 <= wavelength < 645:
        R = 1.0
        G = -(wavelength - 645) / (645 - 580)
        B = 0.0
    elif 645 <= wavelength <= 750:
        R = 1.0
        G = 0.0
        B = 0.0
    
    # 亮度因子
    if 380 <= wavelength < 420:
        factor = 0.3 + 0.7 * (wavelength - 380) / (420 - 380)
    elif 420 <= wavelength < 700:
        factor = 1.0
    elif 700 <= wavelength <= 750:
        factor = 0.3 + 0.7 * (750 - wavelength) / (750 - 700)
    
    return (R * factor, G * factor, B * factor)

def rayleigh_intensity(wavelength, theta_deg):
    """计算瑞利散射强度"""
    theta_rad = np.radians(theta_deg)
    # I ∝ (1 + cos²θ) / λ⁴
    angular_factor = (1 + np.cos(theta_rad)**2)
    wavelength_factor = (450 / wavelength) ** 4  # 归一化到450nm
    return angular_factor * wavelength_factor

class RayleighScatteringDemo:
    def __init__(self):
        self.fig = plt.figure(figsize=(14, 9))
        self.fig.patch.set_facecolor('#0f172a')
        self.fig.suptitle('瑞利散射演示 - 为什么天空是蓝色？(Rayleigh Scattering)',
                         fontsize=16, fontweight='bold', color='white')
        
        # 参数
        self.sun_angle = 30  # 太阳高度角（0=地平线，90=天顶）
        self.view_angle = 90  # 观察角度
        
        # 创建子图
        self.ax_sky = plt.subplot2grid((3, 2), (0, 0), colspan=2, rowspan=2)
        self.ax_spectrum = plt.subplot2grid((3, 2), (2, 0))
        self.ax_polar = plt.subplot2grid((3, 2), (2, 1), projection='polar')
        
        self.create_controls()
        self.update_plot()
        plt.tight_layout()
        plt.show()
    
    def create_controls(self):
        """创建控件"""
        # 太阳高度角滑块
        ax_sun = plt.axes([0.15, 0.08, 0.35, 0.03])
        ax_sun.set_facecolor('#1e293b')
        self.slider_sun = Slider(ax_sun, 'Sun Elevation (°)', 0, 90,
                                valinit=self.sun_angle, color='#fbbf24')
        self.slider_sun.on_changed(self.on_param_changed)
        
        # 观察角度滑块
        ax_view = plt.axes([0.15, 0.04, 0.35, 0.03])
        ax_view.set_facecolor('#1e293b')
        self.slider_view = Slider(ax_view, 'View Angle (°)', 0, 180,
                                 valinit=self.view_angle, color='#22d3ee')
        self.slider_view.on_changed(self.on_param_changed)
    
    def on_param_changed(self, val):
        self.sun_angle = self.slider_sun.val
        self.view_angle = self.slider_view.val
        self.update_plot()
    
    def calculate_sky_color(self, sun_elevation):
        """根据太阳高度计算天空颜色"""
        # 简化模型：低太阳高度角时，蓝光散射损失，红光增加
        path_length_factor = 1.0 / (np.sin(np.radians(sun_elevation + 5)) + 0.1)
        
        wavelengths = np.array([450, 550, 650])  # B, G, R
        intensities = np.array([1.0, 0.6, 0.3])  # 初始强度
        
        # 瑞利散射衰减
        scattered = intensities * rayleigh_intensity(wavelengths, 90)
        transmitted = intensities * np.exp(-path_length_factor * scattered / 10)
        
        # 归一化
        sky_color = scattered / np.max(scattered)
        sun_color = transmitted / np.max(transmitted)
        
        return sky_color, sun_color
    
    def update_plot(self):
        """更新所有图"""
        self.ax_sky.clear()
        self.ax_spectrum.clear()
        self.ax_polar.clear()
        
        for ax in [self.ax_sky, self.ax_spectrum]:
            ax.set_facecolor('#1e293b')
        self.ax_polar.set_facecolor('#1e293b')
        
        # 绘制天空场景
        self.draw_sky_scene()
        
        # 绘制光谱散射强度
        self.draw_spectrum()
        
        # 绘制极坐标散射图
        self.draw_polar_pattern()
        
        self.fig.canvas.draw_idle()
    
    def draw_sky_scene(self):
        """绘制天空场景"""
        self.ax_sky.set_xlim(0, 10)
        self.ax_sky.set_ylim(0, 8)
        self.ax_sky.axis('off')
        
        # 天空颜色（渐变）
        sky_color, sun_color = self.calculate_sky_color(self.sun_angle)
        
        # 背景天空（简化为单色）
        sky_rgb = tuple(sky_color)
        self.ax_sky.add_patch(plt.Rectangle((0, 3), 10, 5,
                                           facecolor=sky_rgb, alpha=0.3))
        
        # 地面
        self.ax_sky.add_patch(plt.Rectangle((0, 0), 10, 3,
                                           facecolor='#2d5016', alpha=0.5))
        
        # 太阳
        sun_rad = np.radians(self.sun_angle)
        sun_x = 2 + 6 * np.cos(sun_rad + np.pi/2)
        sun_y = 3 + 5 * np.sin(sun_rad + np.pi/2)
        
        sun_rgb = tuple(sun_color)
        sun_circle = Circle((sun_x, sun_y), 0.4, color=sun_rgb, alpha=0.9)
        self.ax_sky.add_patch(sun_circle)
        
        # 太阳光线
        for i in range(8):
            angle = i * 45
            ang_rad = np.radians(angle)
            dx = 0.7 * np.cos(ang_rad)
            dy = 0.7 * np.sin(ang_rad)
            self.ax_sky.plot([sun_x, sun_x + dx], [sun_y, sun_y + dy],
                            color=sun_rgb, linewidth=2, alpha=0.7)
        
        # 观察者
        obs_x, obs_y = 5, 3
        obs_circle = Circle((obs_x, obs_y), 0.2, color='white', alpha=0.8)
        self.ax_sky.add_patch(obs_circle)
        self.ax_sky.text(obs_x, obs_y - 0.5, 'Observer', ha='center',
                        color='white', fontsize=9)
        
        # 散射光路
        scatter_x = obs_x + 2 * np.cos(np.radians(self.view_angle))
        scatter_y = obs_y + 2 * np.sin(np.radians(self.view_angle))
        
        arrow = FancyArrowPatch((scatter_x, scatter_y), (obs_x, obs_y),
                               arrowstyle='->', mutation_scale=20,
                               linewidth=2, color='#22d3ee', alpha=0.7)
        self.ax_sky.add_patch(arrow)
        
        # 散射角度标注
        scattering_angle = abs(self.view_angle - self.sun_angle)
        self.ax_sky.text(5, 7, f'Scattering Angle: {scattering_angle:.0f}°',
                        ha='center', color='#22d3ee', fontsize=11,
                        fontweight='bold')
        
        # 偏振信息
        polarization = 100 * np.sin(np.radians(scattering_angle))**2
        self.ax_sky.text(5, 6.5, f'Polarization: {polarization:.0f}%',
                        ha='center', color='#f472b6', fontsize=10)
    
    def draw_spectrum(self):
        """绘制光谱散射强度"""
        wavelengths = np.linspace(380, 750, 100)
        intensities = rayleigh_intensity(wavelengths, self.view_angle)
        
        # 绘制光谱条
        for i, wl in enumerate(wavelengths[:-1]):
            color = wavelength_to_rgb(wl)
            intensity = intensities[i]
            self.ax_spectrum.bar(wl, intensity, width=4, color=color, alpha=0.8)
        
        # 标注关键波长
        blue_wl = 450
        red_wl = 650
        blue_int = rayleigh_intensity(blue_wl, self.view_angle)
        red_int = rayleigh_intensity(red_wl, self.view_angle)
        ratio = blue_int / red_int
        
        self.ax_spectrum.plot(blue_wl, blue_int, 'o', color='blue', markersize=10)
        self.ax_spectrum.plot(red_wl, red_int, 'o', color='red', markersize=10)
        
        self.ax_spectrum.set_xlabel('Wavelength (nm)', color='white')
        self.ax_spectrum.set_ylabel('Scattering Intensity', color='white')
        self.ax_spectrum.set_title(f'Blue/Red Ratio: {ratio:.2f}×', color='white', fontsize=10)
        self.ax_spectrum.tick_params(colors='white')
        self.ax_spectrum.grid(True, alpha=0.2, color='white')
        for spine in self.ax_spectrum.spines.values():
            spine.set_color('white')
    
    def draw_polar_pattern(self):
        """绘制极坐标散射图案"""
        theta = np.linspace(0, 2*np.pi, 100)
        
        # 瑞利散射角度分布：I ∝ 1 + cos²θ
        r = 1 + np.cos(theta)**2
        
        self.ax_polar.plot(theta, r, color='#22d3ee', linewidth=2.5)
        self.ax_polar.fill(theta, r, color='#22d3ee', alpha=0.2)
        
        # 标注90°散射（最大偏振）
        self.ax_polar.plot([np.pi/2, np.pi/2], [0, 1 + np.cos(np.pi/2)**2],
                          'r--', linewidth=2)
        self.ax_polar.plot([3*np.pi/2, 3*np.pi/2], [0, 1 + np.cos(3*np.pi/2)**2],
                          'r--', linewidth=2)
        
        self.ax_polar.set_title('Scattering Pattern\n(1 + cos²θ)', color='white',
                               fontsize=10, pad=20)
        self.ax_polar.grid(True, alpha=0.3, color='white')
        self.ax_polar.tick_params(colors='white')

if __name__ == '__main__':
    print("="*60)
    print("瑞利散射演示 - 为什么天空是蓝色？")
    print("Rayleigh Scattering - Why is the Sky Blue?")
    print("="*60)
    print("\n物理原理:")
    print("  I(θ, λ) ∝ (1 + cos²θ) / λ⁴")
    print("  蓝光散射强度 ≈ 红光的 5.6倍\n")
    
    demo = RayleighScatteringDemo()
